external/avro/src/test/scala/org/apache/spark/sql/avro/AvroLogicalTypeSuite.scala

```diff
@@ -348,6 +348,100 @@ abstract class AvroLogicalTypeSuite extends QueryTest with SharedSparkSession {
       assert(msg.contains("Unscaled value too large for precision"))
     }
   }
+
+  private def readResourceAvroFile(name: String): DataFrame = {
+    val url = Thread.currentThread().getContextClassLoader.getResource(name)
+    spark.read.format("avro").load(url.toString)
+  }
+
+  test("SPARK-31183: compatibility with Spark 2.4 in reading dates/timestamps") {
```

Comment:
missed one thing. I think the test is not very related to logical types and probably should be put in `AvroSuite`.

@MaxGekk can you move the test in your next PR?

Reply:
Do you mean only this test, correct?

Reply:
All the new tests added here. The are more about compatibility, not logical type.